Source/SpectrumAnalysis/KTSeqTrackFinder.cc

```diff
@@ -0,0 +1,539 @@
+    	    }
+
+
+            if (Freq == 0.0 or PointIt->fAmplitude==0.0)
+            {
+                KTDEBUG(stflog, "Point frequency and/or amplitude is zero, skipping point");
+                continue;
+            }
+            else
+            {
+                match = false;
+
+                // loop over active lines, in order of earliest start time
+                // dont need to sort them because they are already sorted by the slice of the line's start point
+
+                //KTDEBUG(stflog, "Currently there are N active Lines "<<fActiveLines.size());
+
+                std::vector< LineRef >::iterator LineIt = fActiveLines.begin();
+                while( LineIt != fActiveLines.end())
+                {
+                    // Under these conditions a point will be added to a line
+                    bool condition1 = PointIt->fTimeInRunC > LineIt->fEndTimeInRunC;
+                    bool condition2 = std::abs(PointIt->fPointFreq - (LineIt->fEndFrequency + LineIt->fSlope*(PointIt->fTimeInAcq - LineIt->fEndTimeInAcq))) < fFrequencyAcceptance;
+
+                    // Check whether line should still be active. If not then check whether the line is a valid new track candidate.
+                    if (LineIt->fEndTimeInRunC<PointIt->fTimeInRunC-fTimeGapTolerance)
+                    {
+                        if (LineIt->fNPoints > fMinPoints)
+                        {
+                            LineIt->LineTrimming(fTrimmingFactor, fMinPoints);
+
+                            if (LineIt->fNPoints > fMinPoints and LineIt->fSlope > fMinSlope)
+                            {
+                                KTINFO(stflog, "Found track candidate");
+                                this->EmitPreCandidate(*LineIt);
+                            }
+                        }
+                        // in any case, this line should be removed from the vector with active lines
+                        LineIt = fActiveLines.erase(LineIt);
+                    }
+                    // if point matches this line: attach
+                    else if (condition1 and condition2)
+                    {
+                        LineIt->InsertPoint(*PointIt);
+                        match = true;
+                        LineIt++;
+                    }
+                    // if not try next line
+                    else
+                    {
+                        LineIt++;
+                    }
+                }
+
+                // if point was not picked up
+                if (match == false)
+                {
+                    //KTDEBUG(stflog, "Starting new line");
+
+                    LineRef new_Line(fInitialSlope);
+                    new_Line.InsertPoint(*PointIt);
+                    fActiveLines.push_back(new_Line);
+                    match = true;
+                }
+            }
+         }
+         return true;
+     }
+
+    bool KTSeqTrackFinder::EmitPreCandidate(LineRef Line)
```

Comment:
Variable names that need to start with lower-case letters: `Line`, `LineIsTrack`